src/main/flight/imu.c

```diff
@@ -342,3 +342,8 @@ int16_t calculateThrottleAngleCorrection(uint8_t throttle_correction_value)
         angle = 900;
     return lrintf(throttle_correction_value * sinf(angle / (900.0f * M_PIf / 2.0f)));
 }
+
+int16_t calculateThrottleAngleCorrectionAxis(uint8_t throttle_hovering_percent, uint8_t axis){
+    uint16_t liftOffTrust = throttle_hovering_percent * 10; //to bring from 0-100 range to 0-1000 range (thrust input is in the 1000-2000 range)
+    return lrintf( liftOffTrust*sinf(anglerad[axis]) );
```

Comment:
This doesn't seem quite right. On the max deflection (90 degrees) you will be adding liftOffTrust, which should not suffice to keep the same downward thrust, as it will translate to 0 downwards. I would expect this to be infinite for 90 degrees and a logic to detect when it's time to abandon the throttle adjustment.
I'm quite confident, that you don't even need throttle_hovering_percent. If the idea is to keep downwards thrust the same independent of the angle, you just need to divide the throttle value (in percents, as in rcCommand, I believe) by the cos of the angle. You definitely need to limit the angle to some feasible value.